pkg/service/service_test.go

```diff
@@ -822,6 +822,11 @@ func (m *ManagerTestSuite) TestUpsertServiceWithTerminatingBackends(c *C) {
 		Name:                      lb.ServiceName{Name: "svc1", Namespace: "ns1"},
 	}
 
+	// Reset state as backends are pointers to lb.Backend
+	p.Backends[0].State = lb.BackendStateActive
+	p.Backends[1].State = lb.BackendStateActive
+	p.Backends[2].State = lb.BackendStateActive
+
```

Comment:
Since backends are passed as pointers now, I would suggest making a deep copy of the test backend objects in unit tests, instead of resetting individual members of the global backend objects in every test case.
There are many instances of `DeepCopy` being called in tests files in the cilium repo in case you were looking for a reference.

Reply:
ok thanks, that make sense